File: genefunk/src/main/java/de/hipp/pnp/genefunk/GeneFunkGenomeRepository.java
package de.hipp.pnp.genefunk;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
interface GeneFunkGenomeRepository extends JpaRepository<GeneFunkGenome, Integer> {
	Optional<GeneFunkGenome> findByName(String name);
}

File: genefunk/src/main/java/de/hipp/pnp/genefunk/GeneFunkGenomeRepository.kt
package de.hipp.pnp.genefunk

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface GeneFunkGenomeRepository : JpaRepository<GeneFunkGenome?, Int?> {
    fun findByName(name: String?): Optional<GeneFunkGenome?>?
}
